finat/argyris.py

```diff
@@ -1,42 +1,83 @@
 import numpy
+from math import comb
 
 import FIAT
 
-from gem import Literal, ListTensor
+from gem import Literal, ListTensor, partial_indexed
 
 from finat.fiat_elements import ScalarFiatElement
 from finat.physically_mapped import PhysicallyMappedElement, Citations
 
 
+def _edge_transform(V, voffset, fiat_cell, moment_deg, coordinate_mapping, avg=False):
+    """Basis transformation for integral edge moments."""
+
+    J = coordinate_mapping.jacobian_at([1/3, 1/3])
+    rns = coordinate_mapping.reference_normals()
+    pts = coordinate_mapping.physical_tangents()
+    pns = coordinate_mapping.physical_normals()
+    pel = coordinate_mapping.physical_edge_lengths()
+
+    top = fiat_cell.get_topology()
+    eoffset = 2 * moment_deg - 1
+    toffset = moment_deg - 1
+    for e in sorted(top[1]):
+        nhat = partial_indexed(rns, (e, ))
+        n = partial_indexed(pns, (e, ))
+        t = partial_indexed(pts, (e, ))
+        Bn = J @ nhat / pel[e]
+        Bnt = Bn @ t
+        Bnn = Bn @ n
+        if avg:
+            Bnn = Bnn * pel[e]
+
+        v0id, v1id = (v * voffset for v in top[1][e])
+        s0 = len(top[0]) * voffset + e * eoffset
+        for k in range(moment_deg):
+            s = s0 + k
+            P1 = Literal(comb(k + 4, k))
+            P0 = -(-1)**k * P1
+            V[s, s] = Bnn
+            V[s, v1id] = P1 * Bnt
+            V[s, v0id] = P0 * Bnt
+            if k > 0:
+                V[s, s + toffset] = -1 * Bnt
+
+
 class Argyris(PhysicallyMappedElement, ScalarFiatElement):
-    def __init__(self, cell, degree):
-        if degree != 5:
-            raise ValueError("Degree must be 5 for Argyris element")
+    def __init__(self, cell, degree, variant=None, avg=False):
         if Citations is not None:
             Citations().register("Argyris1968")
-        super().__init__(FIAT.QuinticArgyris(cell))
+        if variant is None:
+            variant = "integral"
+        if variant == "integral":
+            fiat_element = FIAT.Argyris(cell, degree, variant=variant)
+        elif variant == "point":
+            if degree != 5:
+                raise ValueError("Degree must be 5 for 'point' variant of Argyris")
+            fiat_element = FIAT.QuinticArgyris(cell)
```

Comment:
I thought the point variant with degree 5 in FIAT would be used.  I think we should just have `Argyris`.  
Also, shouldn't we have `NotImplemented` instead -- I think the FIAT element is fine, we just haven't done the transformation theory for point values.